lib/Doctrine/Deprecations/Deprecation.php

```diff
@@ -46,8 +46,8 @@ class Deprecation
     private const TYPE_TRIGGER_ERROR      = 2;
     private const TYPE_PSR_LOGGER         = 4;
 
-    /** @var int */
-    private static $type = self::TYPE_NONE;
+    /** @var self::TYPE_* */
```

Comment:
technically, the type here is `self::TYPE_*|null` now

Reply:
not really because null is never used as a value, it's just the uninitialized state

Reply:
And the uninitialized state for an untyped property is `null`. ☝🏻🤓

Reply:
this discussion is epic :)
describing this as `|null` would be just wrong an inaccurate, especially from a SA pov: no code in there should assume null is a possible value

Reply:
Well, all code in that class has to assume it, which is why you changed it to use `$type = self::$type ?? self::getTypeFromEnv()` and then using `$type` instead of using `self::$type`

Reply:
the `??` operator works with uninitialized properties ;)

Reply:
Ok. That's still a wrong from a type description pov :)

Note that I'm AFK for the weekend and won't be able to move this forward in the very next days.

Feel free to take over if anyone is up to!

Reply:
type updated